rosbag2_cpp/src/rosbag2_cpp/info.cpp

```diff
@@ -52,4 +59,120 @@ rosbag2_storage::BagMetadata Info::read_metadata(
   return storage->get_metadata();
 }
 
+namespace
+{
+struct client_id_hash
+{
+  std::size_t operator()(const std::array<uint8_t, 16> & client_id) const
+  {
+    std::hash<uint8_t> hasher;
+    std::size_t seed = 0;
+    for (const auto & value : client_id) {
+      // 0x9e3779b9 is from https://cryptography.fandom.com/wiki/Tiny_Encryption_Algorithm
+      seed ^= hasher(value) + 0x9e3779b9 + (seed << 6) + (seed >> 2);
+    }
+    return seed;
+  }
+};
+
+using client_id = std::array<uint8_t, 16>;
+using sequence_set = std::unordered_set<int64_t>;
```

Comment:
Comments from MichaelOrlov

```suggestion

using client_id = service_msgs::msg::ServiceEventInfo::_client_gid_type;
using sequence_set = std::unordered_set<int64_t>;
```

Reply:
Updated